axum/src/extract/matched_path.rs

```diff
@@ -82,130 +123,181 @@ where
     }
 }
 
+#[derive(Clone, Debug)]
+struct MatchedNestedPath(Arc<str>);
+
+pub(crate) fn set_matched_path_for_request(
+    id: RouteId,
+    route_id_to_path: &HashMap<RouteId, Arc<str>>,
+    extensions: &mut http::Extensions,
+) {
+    let matched_path = if let Some(matched_path) = route_id_to_path.get(&id) {
+        matched_path
+    } else {
+        #[cfg(debug_assertions)]
+        panic!("should always have a matched path for a route id");
+    };
+
+    let matched_path = append_nested_matched_path(matched_path, extensions);
+
+    if matched_path.contains(NEST_TAIL_PARAM_CAPTURE) {
+        extensions.insert(MatchedNestedPath(matched_path));
+        extensions.remove::<MatchedPath>();
```

Comment:
Why would there be a `MatchedPath` in the extensions in this branch?

Reply:
There shouldn't be. It was mostly a "better safe than sorry". I've changed it to a `debug_assert!`.